dispatch/dispatch.go

```diff
@@ -526,7 +525,7 @@ func (ag *aggrGroup) flush(notify func(...*types.Alert) bool) {
 				level.Error(ag.logger).Log("msg", "failed to get alert", "err", err, "alert", a.String())
 				continue
 			}
-			if a.Resolved() && got.UpdatedAt == a.UpdatedAt {
+			if a.ResolvedAt(now) && got.UpdatedAt == a.UpdatedAt {
```

Comment:
That seems fine. If `got` has any modification reflected in UpdatedAt then we do not remove the alert.